### PR TITLE
Refactor SMTX_NewEqns, SMTX_NullspaceEqns

### PR DESCRIPTION
**Pull request by fingolfin**
Now uses MatrixObj APIs

Motivated by some performance tuning work I did on the MeatAxe some weeks ago for an example by @danielrademacher ; that will come in a separate PR, but for some reasons I also made this cleanup, and it seems a pity to discard it.

### CODE REVIEW

File: tst/testinstall/meatauto.tst [hunk 1]
```diff
@@ -20,7 +20,7 @@ rec( dim := 3, failed := false, field := GF(2), index := [  ],
   mat := [ <a GF2 vector of length 3>, <a GF2 vector of length 3> ], 
   vec := [ 0*Z(2), Z(2)^0 ], weights := [ 1, 2 ] )
 gap> SMTX_NullspaceEqns(e);
-[ [ Z(2)^0, 0*Z(2), Z(2)^0 ] ]
+[ <a GF2 vector of length 3> ]
```
Comment (danielrademacher) [inline tst/testinstall/meatauto.tst:23]: Is it possible to see the "GF2 vector of length 3"? Since the tests worked, they are most probably the same.

**Review by danielrademacher — APPROVED** [review]
Looks good. A lot of improvements if one knows how to work with matrices in GAP. 👍

File: tst/testinstall/meatauto.tst (hunk 1 above)
Comment (fingolfin) [reply tst/testinstall/meatauto.tst:23]: Good point, I've added a test to check the return value.

Comment (danielrademacher) [reply tst/testinstall/meatauto.tst:23]: Great, thanks!

**Pull request merged**